src/helpers/filterHelpers.js

```diff
@@ -15,31 +15,16 @@ const filterProductsUptoPriceRange = (data, maxPriceRange) => {
     return data.filter(product => product.price < maxPriceRange);
 }
 
-const categorizeData = (data, categoryWeights,
-    categorySupplements,
-    categoryEquipments,
-    categoryAccessories) => {
+const categorizeData = (data, categories) => {
 
-    let returnValue = [];
-
-    if (!categoryAccessories && !categorySupplements && !categoryEquipments && !categoryWeights) {
+    if (categories.length < 1) {
         return data;
     }
 
-    if (categoryAccessories) {
-        returnValue = [...returnValue].concat(data.filter(p => p.categoryName === "accessories"));
-    }
-
-    if (categorySupplements) {
-        returnValue = [...returnValue].concat(data.filter(p => p.categoryName === "supplements"));
-    }
-
-    if (categoryWeights) {
-        returnValue = [...returnValue].concat(data.filter(p => p.categoryName === "weights"));
-    }
+    let returnValue = [];
 
-    if (categoryEquipments) {
-        returnValue = [...returnValue].concat(data.filter(p => p.categoryName === "equipments"));
+    for (let category of categories) {
+        returnValue = [...returnValue].concat(data.filter(p => p.categoryName === category));
     }
```

Comment:
Better way pf writing this would be:-
`returnValue = data.filter(item => categories.includes(item.categoryName));`
And it would give you all the products belonging to that category. Try to use inbuilt array functions instead of using for.